### PR TITLE
wdio-browserstack-service: Add configuration to modify session url fo…

### PR DESCRIPTION
**Pull request by garethleonard**
…r appium app-automate

## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
When using the browserstack service with Appium, the "App Automate" session url is different to the "Automate" session url. This change adds a configuration to allow the user to modify this url.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@garethleonard is it possible to detect Appium jobs? If so we could just set the flag based on the capabilities, no?

**Comment by christian-bromann** [comment]
ping @garethleonard

File: packages/wdio-browserstack-service/tests/service.test.js [hunk 1]
```diff
@@ -236,8 +313,7 @@ describe('_getBody', () => {
     it('should return "error" if failures', () => {
         service.failures = 1
         service.failReason = 'error message'
-        service.fullTitle = 'foo - bar',
-
+        ;(service.fullTitle = 'foo - bar'),
```
Comment (CrispusDH) [inline packages/wdio-browserstack-service/tests/service.test.js]: `;` symbol is redundant here

Comment (CrispusDH) [inline packages/wdio-browserstack-service/tests/service.test.js]: could you tell me why you add these symbols `;(    ),` here? :)

Comment (garethleonard) [reply packages/wdio-browserstack-service/tests/service.test.js]: I didn’t notice they were there, could be my VS Code settings? I’ll fix it.

**Comment by garethleonard** [comment]
That could be possible, and would be preferred I think. I’ll look into how browserstack determines that.

**Comment by christian-bromann** [comment]
> I’ll look into how browserstack determines that.

Just check the capabilities if you find Appium specific ones there and set the flag.

**Comment by khause** [comment]
I'm still getting this issue and it's due to `global.browser.capabilities.app` not being an actual property. The `app` capability location is `global.browser.config.capabilities.app`... unless my setup is causing it to be different.

**Pull request merged**